provider/gateway/client.go

```diff
@@ -144,5 +149,88 @@ func makeURI(host string, path string) (string, error) {
 	if err != nil {
 		return "", err
 	}
+
 	return endpoint.String(), nil
 }
+
+func (c *client) ServiceLogs(ctx context.Context,
+	host string,
+	id mtypes.LeaseID,
+	service string,
+	follow bool,
+	tailLines *int64) (*cluster.ServiceLogs, error) {
+
+	type kv struct {
+		k string
+		v string
+	}
+
+	var query []kv
+
+	query = append(query, kv{
+		k: "follow",
+		v: strconv.FormatBool(follow),
+	})
+
+	if tailLines != nil {
+		query = append(query, kv{
+			k: "tail",
+			v: strconv.FormatInt(*tailLines, 10),
+		})
+	}
+
+	uri, err := makeURI(host, serviceLogsPath(id, service))
+	if err != nil {
+		return nil, err
+	}
+
+	for i, kv := range query {
+		if i == 0 {
+			uri += "?"
+		} else {
+			uri += "&"
+		}
+
+		uri += fmt.Sprintf("%s=%s", url.QueryEscape(kv.k), url.QueryEscape(kv.v))
+	}
+
+	conn, _, err := websocket.DefaultDialer.DialContext(ctx, uri, nil)
```

Comment:
:thinking: will cancelling this context shut down the connection, or is it only for dialing?